server/src/test/java/org/elasticsearch/TransportVersionTests.java

```diff
@@ -27,6 +27,17 @@
 
 public class TransportVersionTests extends ESTestCase {
 
+    /**
+     * This test is specific for V9, to ensure that transport versions are backported correctly.
+     */
+    public void testMaximumAllowedTransportVersion() {
+        assertThat(
+            TransportVersions.LATEST_DEFINED.onOrBefore(TransportVersions.ELASTICSEARCH_9_0)
```

Comment:
What is this check achieving? The latest will never be before a known transport constant, and if it would only be equal if there are no patches yet, but isPatchFrom would return true in that case anyways, so I think this is redundant.